### ISSUE TITLE
Question about KLLoss.

### ISSUE THREAD

**Issue by Prasanna1991** [wiseodd/generative-models#29]
Hi,

I am a torch user and want to go toward PyTorch. I am asking the question in reference to conditional vae example written in pytorch. We want errors resulting from KL Loss to backpropagate only through the encoder section. In torch, I specifically do that but here I see you calculating loss (both marginal_liklihood and KLLoss) together and simply doing loss.backward(). How does this ensure that KL loss will backprogate only through the encoder? So does this mean, if I have to add any regularizer, I simply calculate loss and do loss.backward() and errors will back propagate through the desired portion within the network? 

Thanks.

**Comment by wiseodd** [comment]
Hi,
If you look at the KL term, it only depends on `Q(X)`, therefore it will only backprop to `Q(X)`.
Thus calling `loss.backward()` will work just fine because the gradient w.r.t. `P(z)` is 0.

**Comment by Prasanna1991** [comment]
Thanks.
